ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt

```diff
@@ -139,31 +149,34 @@ public class CIOApplicationEngine(environment: ApplicationEngineEnvironment, con
         }
     }
 
-    private fun startConnector(port: Int): HttpServer {
+    private fun CoroutineScope.startConnector(host: String, port: Int): HttpServer {
         val settings = HttpServerSettings(
+            host = host,
             port = port,
             connectionIdleTimeoutSeconds = configuration.connectionIdleTimeoutSeconds.toLong()
         )
 
-        val server = scope.httpServer(settings) { request ->
-            withContext(userDispatcher) {
-                val call = CIOApplicationCall(
-                    application, request, input, output,
-                    engineDispatcher, userDispatcher, upgraded,
-                    remoteAddress,
-                    localAddress
-                )
-
-                try {
-                    pipeline.execute(call)
-                } catch (error: Exception) {
-                    handleFailure(call, error)
-                } finally {
-                    call.release()
-                }
-            }
+        return httpServer(settings) { request ->
+            requestHandler(request)
         }
+    }
+
+    private suspend fun ServerRequestScope.requestHandler(request: Request) {
+        withContext(userDispatcher) {
+            val call = CIOApplicationCall(
+                application, request, input, output,
```

Comment:
codestyle

Reply:
not sure here, but I think each parameter should be on a new line

Reply:
fixed